facade/service.go

```diff
@@ -213,15 +213,19 @@ func (f *Facade) MigrateServices(ctx datastore.Context, request dao.ServiceMigra
 		}
 	}
 
-	if err = f.validateClonedMigrationServices(ctx, request.Cloned); err != nil {
+	if err = f.validateAddedMigrationServices(ctx, request.Added); err != nil {
+		return err
+	}
+
+	if err = f.validateServiceDeploymentRequests(ctx, request.Deploy); err != nil {
 		return err
 	}
 
 	// If this isn't a dry run, make the changes.
 	if !request.DryRun {
 
-		// Add the cloned services.
-		for _, svc := range request.Cloned {
+		// Add the added services.
```

Comment:
This code matches the order from before, but I don't understand: why don't we want to apply our existing state BEFORE we add the new services? Or rather, why is it safe to do it in this order?